src/mapGenerator/mapgenManager.py

```diff
@@ -0,0 +1,156 @@
+
+        actualPath = self.versionController(version)
+
+        if actualPath:
+            auto = Settings.get('mapGenerator/autostart', default=False, type=bool)
+            if not auto:
+                msgbox = QtWidgets.QMessageBox()
+                msgbox.setWindowTitle("Generate map")
+                msgbox.setText("Seems that you don't have the map used this game. Do you want to generate it?<br/><b>" + mapname + "</b>")
+                msgbox.setInformativeText("Map generation is CPU intensive task and may take some time.")
+                msgbox.setStandardButtons(QtWidgets.QMessageBox.Yes | QtWidgets.QMessageBox.YesToAll | QtWidgets.QMessageBox.No)
+                result = msgbox.exec_()
+                if result == QtWidgets.QMessageBox.No:
+                    return False
+                elif result == QtWidgets.QMessageBox.YesToAll:
+                    Settings.set('mapGenerator/autostart', True)
+            
+            # Start generator with progress bar      
+            self.generatorProcess = MapGeneratorProcess(actualPath, self.mapsFolder, seed, version, mapname)
+            
+            # Check if map exists or generator failed
+            if os.path.isdir(os.path.join(self.mapsFolder, mapname)):
+                if self.previousMaps:
+                    self.previousMaps = self.previousMaps + ";" + mapname
+                else:
+                    self.previousMaps = mapname
+                Settings.set('mapGenerator/deleteMaps', self.previousMaps)
+                
+                return mapname
+            else:
+                return False
+        else:
+            return False
+            
+    def generateRandomMap(self):
+        ''' 
+        Called when user click "generate map" in host widget.
+        Prepares seed and requests latest version once per session
+        '''
+
+        if self.currentVersion == "0" or not self.latestVersion:
+            self.checkUpdates()
+
+            if self.latestVersion and self.versionController(self.latestVersion):
+                self.currentVersion = self.latestVersion   # mapgen is up-to-date
+                Settings.set('mapGenerator/version', self.currentVersion)
+            elif self.currentVersion == "0":               # if not "0", use older version
+                return False                               # otherwise we don't have any generator at all
+
+        seed = ''.join(["%s" % random.randint(0, 9) for num in range(0, 19)])
+        mapName = "neroxis_map_generator_{}_{}".format(self.currentVersion, seed)
+
+        return self.generateMap(mapName)
+   
+    def versionController(self, version):
+        name = "NeroxisGen_{}.jar".format(version)
+        filePath = self.generatorPath + "\\" + name
+        
+        # Check if required version is already in folder
+        if os.path.isdir(self.generatorPath):
+            for infile in os.listdir(self.generatorPath):
+                if infile.lower() == name.lower():
```

Comment:
why are you lowercasing here?

Reply:
Does the map generator preserve case everywhere? I imagine this is trying to be on the safe side, since there's multiple maps that mix case between zips, directories and files. Windows doesn't care, linux does.